proposals/NNNN-isolated-any-functions.md

```diff
@@ -0,0 +1,671 @@
+assumed to always cross an isolation boundary.  This means, among other
+things, that the call is effectively asynchronous and must be `await`ed.
+
+```swift
+await operation(timePassed / overallDuration)
+```
+
+The isolation can be read using the special `isolation` property
+of these types:
+
+```swift
+func traverse(operation: isolated(any) (Node) -> ()) {
+  let isolation = operation.isolation
+}
+```
+
+The isolation checker knows that the value of this special property
+matches the isolation of the function, so calls to the function from
+contexts that are isolated to the `isolation` value do not cross
+an isolation boundary.
+
+Finally, every task-creation API in the standard library will be updated
+to take a `@isolated(any)` function value and synchronously enqueue the
+new task on the appropriate executor.
+
+## Detailed design
+
+### Grammar and structural rules
+
+`@isolated(any)` is a new type attribute that can only be applied to
+function types.  It is an isolation specification, and it is an error
+to combine it with other isolation specifications such as a global
+actor attribute or an `isolated` parameter.
+
+`@isolated(any)` is not a *concrete* isolation specification and cannot
+be directly applied to a declaration or a closure.  That is, you cannot
+declare a function *entity* as having `@isolated(any)` isolation,
+because Swift needs to know what the actual isolation is, and
+`@isolated(any)` does not provide a rule for that.
+
+To reduce the number of attributes necessary in typical uses,
+`@isolated(any)` implies `@Sendable`.  It is generally not useful
+to use `@isolated(any)` on a non-`Sendable` function because a
+non-`Sendable` function must be isolated to the current concurrent
+context.
+
+### Conversions
+
+Let `F` and `G` be function types, and let `F'` and `G'` be the corresponding
+function types with any isolation specifier removed (including but not
+limited to `@isolated(any)`.  If either `F` or `G` specifies
+`@isolated(any)` then a value of type `F` can be converted to type `G`
+if a value of type `F'` could be converted to type `G'` *and* the
+following conditions apply:
+
+- If `F` and `G` both specify `@isolated(any)`, there are no further
+  conditions.  The resulting function is dynamically isolated to the
+  same value as the original function.
+
+- If only `G` specifies `@isolated(any)`, then the behavior depends on the
+  specified isolation of `F`:
+
+  - If `F` has an `isolated` parameter, the conversion is invalid.
+  - Otherwise, the conversion is valid, and the dynamic isolation of
+    the resulting function is determined as follows:
+    - If the converted value is the result of an expression that is a
+      closure expression (including an implicit autoclosure), a function
+      reference, or a partial application of a method reference, the
+      resulting function is dynamically isolated to the isolation of the
+      function or closure.  This looks through non-instrumental differences
```

Comment:
"non-instrumental differences" isn't a phrase I've ever seen us use. Perhaps "looks though syntax that has no effect on the semantics of the expression, such as parentheses"?